### DISCUSSION TITLE
An idea that's quite out there: Live Linux Distro for Lighthouse

### DISCUSSION THREAD

**Discussion by vilijur** [LBPUnion/ProjectLighthouse#534] [Ideas]
This idea's quite out there, if I'm being honest, but there's a possibility that some people may not be able to use their current operating system to patch their EBOOT. So, here's what I propose...

# A Live Linux Distro used for patching to Lighthouse
It's quite simple: All the user has to do is download the ISO, put it on a USB Drive, boot off of the USB Drive, do the patching in the live environment of the distro, and then once they're done, they can shut down, remove the drive, and boot back into their host operating system.

Since a live environment of Linux allows people to do things without **installing** Linux, it should allow those who are, for some reason, unable to patch their game using the host operating system to patch without affecting their host operating system*, or having to resort to more manual methods.

## The Intended Audience
Now, who could this supposed Linux Distro be used by? Well, it'd probably be best for users with a PS3 (maybe also a PS Vita) who can't get Union Patcher starting up on their machine for whatever reason. Since RPCS3 is a PC application, it may be more of a hassle to use the Linux Distro just to patch their eboot, so it may be best for those players to try and get Union Patcher working on the host machine instead.

## The Contents
Now that I've gone over the general idea of it, what about **the contents?** Well, it may have something along the lines of...

- Union Patcher (and all it's dependencies)
- No way to install the distro (It is a live distro at heart, and installing it defeats it's purpose)
- Wine** (in the event that a Windows application is needed)
- A web browser (For signing in and authenticating. Saves time in case something goes wrong with patching)
- Maybe other stuff I'm forgetting

## The Problems
Now, with all great ideas (I don't think I can call my ideas "great"..) comes flaws that should be addressed.

### Linux and Wi-Fi Cards
These two don't tend to play nice, and it's more then possible that a user can't connect to their Wi-Fi network, and in turn, their PS3. Finding a way to either mitigate this as much as possible would be good, but this is bound to be an issue that may not be easy to fix.

### Linux and x32 bit systems
As 32 bit systems start to get faded out more and more, so do 32 bit OSs. Linux is no exception to this, and being able to avoid this isn't a possibility. If someone with a 32 bit computer as their only computer attempts to patch their game, they may not get far. Though unlikely, it can still happen.

### Some other issue I'm forgetting
idk my brain tired, brain go brrrrr

## The Problems That Get Solved
Problems are all fine and dandy, but if this proposed idea did exist, it could solve the following problems:

- [x] Being unable to install Union Patcher's Dependencies.
- [x] Allowing someone without their own personal computer to patch their game on a separate machine without affecting the computer's host drive* (assuming the owner can trust them enough).
- [x] Not having to use slow boring clunky Windows 10 or shouldn't technically exist but it does anyway so here we are Windows 11
- [x] Others that I'm missing or haven't thought of, probably.

## The Problems that Don't Get Solved
Unfortunately, Live Linux can't solve everything, such as:

- [ ] A User being unable to boot onto a USB Drive for some reason (Unable to change boot order, Chromebooks probably, Mac probably, etc)
- [ ] A User on outdated hardware that can't run Union Patcher **not due to issues installing dependencies** (32 bit operating systems, outdated OS for some reason, old Chromebooks that don't allow you to change the boot order, Mac again probably)
- [ ] An error with Union Patcher itself (a patching error, like being unable to download the file, or being unable to connect to the PS3. That should be troubleshooted on the host machine)
- [ ] Others that I'm missing or haven't thought of, probably.

# Conclusion
The Live Linux distro would be able to patch a copy of LBP1-3 (maybe Vita) on a machine that is unable to run Union Patcher. It contains all the necessary programs needed to patch, and lacks an option to install it, so people don't accidentally modify the hard drive of the machine. It's to be as user friendly as possible, as well as straight to the point to get users to patch, connect, then exit the live environment. It should be used as a last resort, as patching on the host machine should be prioritised, however this should exist as an option for those who need to use it, and have no other choice but it. Since it has everything already set up on it, the user doesn't need to install any additional applications in order to get going, all the user needs is a basic understanding of how to navigate, as well as knowing how to patch using Union Patcher.

...or maybe this is all just over-engineering for a simple problem idk

# Disclaimers and other tidbits that i don't know where to put
This is all a hypothetical idea, and designed for specific use cases only. Although it'd be nice to have one of my ideas be used in production, I'm not too confident in that happening. This will also most likely not get used much due to it's specific use case.

If this were to be made, it'd have to be based off of a Linux distro that works best with Union Patcher, but is also fairly user friendly. Maybe Debian with the Cinnamon DE?

*Tools to make the USB Drive bootable are required, as well as downloading the ISO. That's all that needs to be done on the host.

**The inclusion of Wine is optional. It'll most likely be needed for any Windows applications that are used for patching LBP Vita, and even then, it may be more of a hassle then just patching it on a Windows machine. Also there's no need to have the Windows version of Union Patcher as well as the Linux version lol

**Comment by m88youngling** [comment]
This... Is SO COOL!

**Comment by OrionMelaeno** [comment]
Well... You know what they say; "Practical problems... Require practical solutions."

**Comment by TorutheRedFox** [comment]
ok hear me out

patcher homebrew for the PS3

**Reply by KairosEasyBro** [reply]
writing the url would be uhhhh

**Reply by vilijur** [reply]
You can always connect a USB keyboard to the PS3.

Though, having homebrew that patches the game on the system itself is probably possible. :thinking:
I think this has been discussed before though. Idk my brain dumb

**Reply by TorutheRedFox** [reply]
> writing the url would be uhhhh

less annoying than copying files to and from a PS3 (done that many times, it's annoying af)

**Reply by KairosEasyBro** [reply]
> > writing the url would be uhhhh
> 
> less annoying than copying files to and from a PS3 (done that many times, it's annoying af)

unionpatcher already makes it almost instant, even more with the last ver, that allows you to patch the game even when it's already patched

**Comment by daggintosh** [comment]
we would need to use Linux 5.15 with the proprietary firmware included to support WiFi.

i have no idea if we would need a license for that

**Reply by vilijur** [reply]
yeah, WiFi is going to be quite a problem...

fun linux quirks!
